lib/flop.ex

```diff
@@ -1248,7 +1251,23 @@ defmodule Flop do
     {[field | order_by || []], [new_direction | order_directions || []]}
   end
 
-  defp default_repo, do: Application.get_env(:flop, :repo)
+  defp apply_on_repo(repo_fn, flop_fn \\ nil, args, opts) do
+    repo =
+      option_or_default(opts, :repo) || raise no_repo_error(flop_fn || repo_fn)
```

Comment:
If you apply the change from my comment above, you can remove the default value here.

```suggestion
  defp apply_on_repo(repo_fn, flop_fn, args, opts) do
    repo = option_or_default(opts, :repo) || raise no_repo_error(flop_fn)
```